test/test_pipeline/test_classification.py

```diff
@@ -626,3 +653,171 @@ def test_add_preprocessor(self):
         cs = SimpleClassificationPipeline().get_hyperparameter_search_space()
         self.assertIn('DummyPreprocessor', str(cs))
         del preprocessing_components._addons.components['DummyPreprocessor']
+
+    def _test_set_hyperparameter_choice(self, expected_key, implementation, config_dict):
+        """
+        Given a configuration in config, this procedure makes sure that
+        the given implementation, which should be a Choice component, honors
+        the type of the object, and any hyperparameter associated to it
+        """
+        keys_checked = [expected_key]
+        implementation_type = config_dict[expected_key]
+        expected_type = implementation.get_components()[implementation_type]
+        keys_checked.append(expected_key)
```

Comment:
In the end I compare set() because we are dealing with dict keys, so functionally it is not an issue but clearly doesn't look nice. I have fixed this, thank for pointing this out!